### PR TITLE
feat(suite-native): mark possible address poisoning txns (#7400)

### LINKED ISSUE #7400: Address poisoning attack: Label zero value transactions in TX detail
As a user I would like to be warned that I might have received potential scam transaction in order to avoid copying the address and use it elsewhere and lost money.

When potential scam transaction with Zero value appears, we display icon on transaction list with exclamation mark

When user access the transaction, there is a notification informing user about possible fraudulent with text: "Caution! This transaction may be a scam. If you’re unsure, don't engage. Read more here (link: https://trezor.io/support/a/address-poisoning-attacks). 

Figma: https://www.figma.com/file/xiPjYIgQtWDEhpFBwvhLlj/MobileApp-%E2%80%A2-Assorted?type=design&node-id=2088%3A22475&mode=design&t=dPfd5yhLrPa9lJAK-1

Same issue has been developed in Desktop Suite: [Address poisoning attack: Label zero value transactions in TX detail](https://github.com/trezor/trezor-suite/issues/7278)

### PR DESCRIPTION
**Pull request by vytick**
For potentially poisonous txns:
- added label to txn list
- added warning to txn detail

## Description

Warn user that he might have received potential scam transaction in order to avoid copying the address and use it elsewhere and lost money.

## Related Issue

Closes #7400

## Screenshots:
Simulator Screenshot - iPhone 14 - 2023-10-26 at 16 19 50
Simulator Screenshot - iPhone 14 - 2023-10-26 at 20 46 10

### CODE REVIEW

File: suite-native/intl/src/en.ts
Comment (PeKne) [inline suite-native/intl/src/en.ts]: This translated string is duplicated. Maybe create it just once?

File: suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx
Comment (PeKne) [inline suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx]:
```suggestion
    ...utils.boxShadows.small,
```
The Android and iOS work differently with shadows. This utils styles ensures that the shadows are applied for both platforms. More here, if you are interested: https://blog.logrocket.com/applying-box-shadows-in-react-native/.

File: suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx
Comment (PeKne) [inline suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx]: nit:
```suggestion
    const transaction = useSelector((state: TransactionsRootState) =>
```

We are usually trying to keep the variable name aligned with the selector name.

Comment (PeKne) [reply suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx]: Btw is not this selected transaction exactly the same one, that is already stored in the `transaction` component prop?

File: suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx
Comment (PeKne) [inline suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx]: improvement proposal:

you could create a new selector `selectIsZeroValuePhishingTransaction` and hide selecting of a transaction and evaluating it inside of the selector.

But I don't insist on that. Just an idea.

**Review by PeKne — CHANGES_REQUESTED** [review]
Some nits. Overall looks very good.

File: suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx
Comment (vytick) [reply suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx]: But I want no shadow and there is a default shadow present. By setting the color to `rgba(0, 0, 0, 0)` I effectively make the shadow disappear

Comment (PeKne) [reply suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx]: Ah, I see. My bad.

File: suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx
Comment (vytick) [reply suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx]: good catch. I needed different type, but the selector still needs to be recast.

File: suite-native/intl/src/en.ts
Comment (vytick) [reply suite-native/intl/src/en.ts]: correct. They are the same

I've been thinking about this and my approach is this one:

There might be technically multiple duplicates of some texts. Those texts might be used in different contexts, sometimes as CTA, sometimes as Title etc. But once translations are done on those texts according to the context or language nuances, the texts in the other translations can differ per each context (and in my experience they sometimes do). For that reason, if texts are used in different contexts, they should/can be duplicate. This situation here is somewhere on the edge between the same context and different one, so I was not 100% sure what approach to take.

Wdyt? Let's discuss! :)

File: suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx]: eaa9f838b

File: suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx
Comment (vytick) [reply suite-native/transactions/src/components/TransactionDetail/TransactionDetailData.tsx]: eaa9f838b

File: suite-native/intl/src/en.ts
Comment (PeKne) [reply suite-native/intl/src/en.ts]: I think you are right. The context might change... 

So maybe let's don't use `caution` as a translation string id, and make it whole more comprehensive?
eg:
```
transactions: {
     phishing: {
        badgeLabel: "Caution!"
        alertTitle: "Caution!",
        alertContent: "..."
    }
}
```

What do you think?

File: suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx]: Did you try to utilize this Atom component? `AlertBox.tsx`

Comment (juriczech) [reply suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx]: You should be able to extend the functionality a little bit and just add a new variant.

File: suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx
Comment (juriczech) [inline suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx:191]: You're missing a little bit of left padding according to design. Discuss with Lukas if this is fine. If not, just extend with styles prop on the Badge component to add it there.

**Review by juriczech — CHANGES_REQUESTED** [review]

File: suite-native/intl/src/en.ts
Comment (Nodonisko) [reply suite-native/intl/src/en.ts]: Answer it as always it depends. I will leave this up to you guys, if you think that context might change feel free to duplicate that, but also try to avoid unneccasary duplications if possible.

Also it's worth mentioning that it's very easy to just replace that one usage with changed context with more specific variant than doing opposite.

Also I agree with @PeKne about more general translations ids that are not that tied to it's content, but again it depends...

**Review by PeKne — APPROVED** [review]

File: suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx:191]: resolved via 9cdf0d6

File: suite-native/intl/src/en.ts
Comment (vytick) [reply suite-native/intl/src/en.ts]: 09ace9f

**Review by juriczech — APPROVED** [review]

File: suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx
Comment (Nodonisko) [inline suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx]: https://github.com/trezor/trezor-suite/pull/9791

Comment (vytick) [reply suite-native/transactions/src/components/TransactionDetail/TransactionDetailWarning.tsx]: 17f3735

**Comment by vytick** [comment]
to be finished once #9798 is merged

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**
